compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -527,7 +527,7 @@ class Inliner(call: tpd.Tree, rhs: tpd.Tree)(implicit ctx: Context) {
       cond1.tpe.widenTermRefExpr match {
         case ConstantType(Constant(condVal: Boolean)) =>
           val selected = typed(if (condVal) tree.thenp else tree.elsep, pt)
-          if (isIdempotentExpr(cond1)) selected
+          if (isPureExpr(cond1)) selected
```

Comment:
In fact it looks like all constant type conditions that reach this code are pure. Are they all supposed to be in this particular case? If not how could I construct a condition that has a constant type and is not pure.